lib/cartopy/mpl/geoaxes.py

```diff
@@ -427,10 +427,45 @@ def draw(self, renderer=None, inframe=False):
                 self.imshow(img, extent=extent, origin=origin,
                             transform=factory.crs, *args[1:], **kwargs)
         self._done_img_factory = True
-
+        self._cachedRenderer = renderer
         return matplotlib.axes.Axes.draw(self, renderer=renderer,
                                          inframe=inframe)
 
+    def _update_title_position(self, renderer):
+        """
+        Update the title position based on the bounding box enclosing
+        all the ticklabels and x-axis spine and xlabel...
+
+        """
+        matplotlib.axes.Axes._update_title_position(self, renderer)
+        if not self._gridliners:
+            return
+
+        if self._autotitlepos is not None and not self._autotitlepos:
+            return
+
+        # Get the max ymax of all top labels
+        top = -1
+        for gl in self._gridliners:
+            if gl.has_labels():
+                for label in (gl.top_label_artists +
+                              gl.left_label_artists +
+                              gl.right_label_artists):
+                    bb = label.get_tightbbox(renderer)
+                    top = max(top, bb.ymax)
+        if top < 0:
```

Comment:
Can you add a short comment about why you are returning without adjustment in the case in which `top < 0` and that in which `yn <= 1` please?  It's not very clear.

Reply:
When `top < 0`, the labels are all below the axes upper bound and they basically can't overlap the tile.

Reply:
That makes sense.  A little comment in the code to that effect would be great.